client/src/components/Attachment/UploadAttachment.js

```diff
@@ -45,6 +45,10 @@ const UploadAttachment = ({
     const selectedAttachment = new Attachment({
       fileName: file.name,
       mimeType: file.type,
+      caption:
+        file.name.lastIndexOf(".") !== -1
+          ? file.name.substr(0, file.name.lastIndexOf("."))
+          : file.name,
```

Comment:
First of all I would reverse the condition, so: `index === -1 ? file.name : file.name.substr(0, index)`
Secondly I would try to avoid the repeated expression `file.name.lastIndexOf(".")` (perhaps introduce a util function?)
Thirdly, `substr()` is deprecated, you should use `substring()`